core/whistle_apps-1.0.0/src/whapps_speech.erl

```diff
@@ -290,6 +354,37 @@ create_response(_Engine, {'error', _R}) ->
 create_response(_Engine, {'ibrowse_req_id', ReqID}) ->
     lager:debug("speech file streaming as ~p", [ReqID]),
     {'ok', ReqID};
+create_response(<<"voicefabric">> = _Engine, {'ok', "200", Headers, Content}) ->
+    [lager:debug("hdr: ~p", [H]) || H <- Headers],
+    lager:debug("converting media"),
+    RawFile = tmp_file_name(<<"raw">>),
+    WavFile = tmp_file_name(<<"wav">>),
+    _ = file:write_file(RawFile, Content),
+    From = "raw -r 8000 -e signed-integer -b 16",
```

Comment:
Try applying this diff: https://gist.github.com/jamesaimonetti/62d7267d0265d6eabe12 to wh_media_util.

Regarding your points:
1. I'm confused where sox needs length? I would guess byte-order and encoding would work from stdin too, but maybe not?
2. If 1 isn't addressed with the diff above, we'll roll with this for now.
3. Ok
4. Configurable timeout now in the Options list.

Give it a try and see if it works for you; if not, we'll get this merged and revisit it another time.